### PR TITLE
Add staff to be eligible for library bundle

### PR DESCRIPTION
**Pull request by suecarmol**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Added the ability for all Wikipedia Library staff to access the Library Bundle from the management command `user_update_eligibility`.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
Since Wikipedia Library staff need to test the application's functionality, it is crucial that they have access to the library bundle so they can test it.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T252429](https://phabricator.wikimedia.org/T252429)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
Automatic tests have been added.

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
N/A

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by jsnshrmn** [comment]
I was imagining this as an override that staffers could toggle on/off to temporarily force bundle eligibility. This approach works too, and is a good way to start simple. Since we're adding all staff here, we should also add superusers, since it's been our practice that superusers get everything that's authorized in a role-based way.

**Comment by suecarmol** [comment]
> I was imagining this as an override that staffers could toggle on/off to temporarily force bundle eligibility. This approach works too, and is a good way to start simple. Since we're adding all staff here, we should also add superusers, since it's been our practice that superusers get everything that's authorized in a role-based way.

Thanks for the input! I have added superusers to the bundle authorization as well.

**Comment by suecarmol** [comment]
> Looking through this more closely:
> it seems like staff / superuser check should be made in `TWLight.users.helpers.editor_data`.
> The current code should work fine for the cron task, but we also check eligibility at login. I would expect the check at login time to potentially overwrite the value from the cron run.

Before I start working on that, I think it would be best to merge #494 so I can make those changes based on that refactoring.

**Comment by suecarmol** [comment]
After much thinking instead of adding the condition on `TWLight.users.helpers.editor_data`, I decided to add it to `TWLight.users.models.Editor.update_from_wikipedia` and keep it in the `update_user_eligibility` command. Let me know what you think!

**Comment by jsnshrmn** [comment]
I think this is over-complicating things. We've decided that staff and superusers are eligible for the bundle, and we have a helper function whose sole purpose is determining bundle eligibility. Let's put the logic there. The end state (for the session and what's stored in the db) will be the same, and the code will be more maintainable.

**Comment by jsnshrmn** [comment]
Since this will require changing the signature of `editor_bundle_eligible` to include  a few user (not editor) fields, you could feed it a whole user object instead of the individual fields.

File: TWLight/users/tests.py [hunk 1]
```diff
@@ -1010,28 +1010,12 @@ def test_is_user_valid(self):
         )
         self.assertTrue(valid)
 
-    def test_editor_eligibility_functions(self):
```
Comment (suecarmol) [inline TWLight/users/tests.py:1013]: @jsnshrmn I removed this test function because I think we already test `editor_bundle_eligible` throughout different tests, including the one that I added in this PR. Let me know if you think I should refactor these tests instead of deleting them

**Comment by suecarmol** [comment]
> I think this is over-complicating things. We've decided that staff and superusers are eligible for the bundle, and we have a helper function whose sole purpose is determining bundle eligibility. Let's put the logic there. The end state (for the session and what's stored in the db) will be the same, and the code will be more maintainable.

You are right, I did overthink this solution. I have changed it. Let me know what you think!

File: TWLight/users/management/commands/user_update_eligibility.py [hunk 2]
```diff
@@ -40,7 +41,12 @@ def handle(self, *args, **options):
         if options["datetime"]:
             wp_editcount_updated = datetime.fromisoformat(options["datetime"])
 
-        editors = Editor.objects.filter(wp_bundle_eligible=True)
+        # Getting all editors that are currently eligible or are staff or are superusers
```
Comment (jsnshrmn) [inline TWLight/users/management/commands/user_update_eligibility.py:44]: I would revert this change and use a simple migration to set all staff + superusers to eligible.

File: TWLight/users/helpers/editor_data.py [hunk 3]
```diff
@@ -184,8 +184,11 @@ def editor_recent_edits(
     )
 
 
-def editor_bundle_eligible(wp_valid, wp_enough_recent_edits):
-    if wp_valid and wp_enough_recent_edits:
+def editor_bundle_eligible(editor):
```
Comment (jsnshrmn) [inline TWLight/users/helpers/editor_data.py:187]: This is a much better signature than what I did. Thanks!

File: TWLight/users/tests.py (hunk 1 above)
Comment (jsnshrmn) [reply TWLight/users/tests.py:1013]: Yep, we have some redundancy there. The deletions are a-okay. We can just leave it at that since we have a big test redesign in our future.

File: TWLight/users/management/commands/user_update_eligibility.py (hunk 2 above)
Comment (suecarmol) [reply TWLight/users/management/commands/user_update_eligibility.py:43]: Got it! I am removing some test cases I added, since this will break them (but I don't think we'll have an actual problem in production because of the migration I created.
